shared/src/Datadog.Trace.ClrProfiler.Native/single_step_guard_rails.cpp

```diff
@@ -286,18 +291,52 @@ void SingleStepGuardRails::SendTelemetry(const std::string& runtimeName, const s
 
     Log::Debug("SingleStepGuardRails::SendTelemetry: Invoking: ", processPath, " with ", initialArg, "and metadata " , metadata);
 
-    std::thread([processPath, args, metadata]()
+    // Increment the reference count to prevent the loader from being unloaded while sending telemetry
+
+#ifdef _WIN32
+    HMODULE handle;
+
+    // GetModuleHandleEx increments the reference count if called without GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT
+    if (!GetModuleHandleEx(GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS,
+        (LPCWSTR)&GetRuntimeId, &handle))
     {
-        const auto success = ProcessHelper::RunProcess(processPath, args, metadata);
+        handle = 0;
+    }
 
-        if (success)
-        {
-            Log::Debug("SingleStepGuardRails::SendTelemetry: Telemetry sent to forwarder");
-        }
-        else
+#else
+    // Use dlopen to increment the reference count of the module
+    void* handle = 0;
+    Dl_info info;
+    if (dladdr((void*)&GetRuntimeId, &info))
+    {
+        handle = dlopen(info.dli_fname, RTLD_LAZY);
+    }
+#endif
```

Comment:
I think yes but are we sure this works as intended on macOS?

Do we need some tests?

Reply:
OSX has `dlopen` so that _should_ work 🤔